portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,76 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobInfo;
+import com.google.appengine.api.blobstore.BlobInfoFactory;
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.images.ImagesService;
+import com.google.appengine.api.images.ImagesServiceFactory;
+import com.google.appengine.api.images.ServingUrlOptions;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.sps.data.Comment;
+import com.google.sps.data.MemePost;
+
+@WebServlet("/form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String description = request.getParameter("description-input");
+        String imageUrl = getImageUrl(request, "image-input");
+        MemePost memePost = new MemePost("Test-Name", description, imageUrl);
+        memePost.putInDatastore();
+        response.sendRedirect("/image-upload.html");
+    }
+
+    /**
+    * Returns a valid image url string  
+    */
+    public String getImageUrl(HttpServletRequest request, String formName) throws IOException{
```

Comment:
Its a good idea to run code through a formatter before commiting (there is not whitespace between IOException and the prace)

Reply:
Acknowledged.